### PR TITLE
Add test for dev portal sections

### PR DESCRIPTION
**Pull request by JaurbanRH**

### CODE REVIEW

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    custom_devel_login(account=account)
+    browser.url = testconfig["threescale"]["devel"]["url"] + dev_portal_page
+
+    assert browser.element(".//h1").accessible_name == "Test"
+
+    custom_devel_login(name="john", password="123456", fresh=True)
+    browser.url = testconfig["threescale"]["devel"]["url"] + dev_portal_page
+
+    assert browser.elements(".//h1")[1].accessible_name == "Not Found"
```
Comment (dhlavac) [inline testsuite/tests/ui/devel/test_devel_sections.py:98]: Cannot we use not found page view?

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+# pylint: disable=unused-argument, too-many-arguments
+
+
+@pytest.fixture()
+def dev_portal_section(login, navigator, request, custom_admin_login):
+    """Creates a new section in developer portal"""
+    view = navigator.navigate(DeveloperPortalContentNewSectionView)
+    section_name = blame(request, "TestSection")
+    section_path = blame(request, "/section")
```
Comment (dhlavac) [inline testsuite/tests/ui/devel/test_devel_sections.py:21]: the section should be without starting slash

**Review by dhlavac — CHANGES_REQUESTED** [review]

File: testsuite/ui/views/admin/audience/account.py [hunk 3]
```diff
@@ -84,6 +85,11 @@ def invoices(self):
         """Open account's users"""
         self.invoices_button.click()
 
+    @step("AccountUserGroupView")
+    def group(self):
+        """Edit account"""
```
Comment (dhlavac) [inline testsuite/ui/views/admin/audience/account.py:90]: Edit groups ?

File: testsuite/ui/views/admin/audience/developer_portal/__init__.py [hunk 4]
```diff
@@ -2,19 +2,151 @@
...
+
+class DeveloperPortalContentNewSectionView(BaseAudienceView):
+    """View representation of Developer Portal New Section page"""
+    path_pattern = '/p/admin/cms/sections/new'
+    title = TextInput(id='cms_section_title')
+    public = ThreescaleCheckBox('//input[@id="cms_section_public"]')
+    path_input = TextInput(id='cms_section_partial_path')
+    submit = ThreescaleSubmitButton()
+
+    def create(self, title, path, public=False):
```
Comment (dhlavac) [inline testsuite/ui/views/admin/audience/developer_portal/__init__.py:81]: the default value should be True, private sections are only corner cases

File: testsuite/ui/widgets/__init__.py [hunk 5]
```diff
@@ -441,3 +458,29 @@ def text(self):
...
+
+class ThreescaleButtonGroup(GenericLocatorWidget):
+    """Specific Button Group of 3scale pages"""
+
+    SELECTED = "./a[1]"
+    DROPDOWN_ITEMS = "./ul/li/a"
+    DROPDOWN_TOGGLE = "./a[2]"
+
+    def __init__(self, parent=None,
+                 locator=".//*[@id='cms-new-content-button']",
```
Comment (dhlavac) [inline testsuite/ui/widgets/__init__.py:471]: I would suggest passing the locator to it, this default locator is specific for only one button group in 3scale in the developer portal section

**Review by dhlavac — CHANGES_REQUESTED** [review]

**Review by dhlavac — APPROVED** [review]

File: testsuite/ui/views/admin/audience/developer_portal/__init__.py [hunk 6]
```diff
@@ -2,19 +2,151 @@
...
+    def delete(self):
+        """Delete page of dev portal"""
+        self.delete_button.click()
+
+    def prerequisite(self):
+        return DeveloperPortalContentView
+
+    @property
+    def is_displayed(self):
+        return BaseAudienceView.is_displayed.fget(
```
Comment (jsmolar) [inline testsuite/ui/views/admin/audience/developer_portal/__init__.py:68]: line separation

File: testsuite/ui/views/admin/audience/developer_portal/__init__.py [hunk 7]
```diff
@@ -2,19 +2,151 @@
...
 
 from testsuite.ui.navigation import step
 from testsuite.ui.views.admin.audience import BaseAudienceView
-from testsuite.ui.widgets import ThreescaleDropdown, DivBasedEditor
+from testsuite.ui.widgets import ThreescaleDropdown, DivBasedEditor, ThreescaleButtonGroup, ThreescaleCheckBox, \
+    CheckBoxGroup
 from testsuite.ui.widgets.buttons import ThreescaleSubmitButton, ThreescaleDeleteButton, ThreescaleCreateButton
 
 
+class DeveloperPortalContentNewPageView(BaseAudienceView):
```
Comment (jsmolar) [inline testsuite/ui/views/admin/audience/developer_portal/__init__.py:14]: The naming of these classes is a bit off. I would opt for something like `CMSNewPageView` or `ContentNewPageView`

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    """
+    custom_devel_login(account=account)
+    browser.url = testconfig["threescale"]["devel"]["url"] + dev_portal_page
+
+    assert browser.element(".//h1").accessible_name == "Test"
+
+    custom_devel_login(name="john", password="123456", fresh=True)
+    browser.url = testconfig["threescale"]["devel"]["url"] + dev_portal_page
+
+    not_found = navigator.new_page(NotFoundView)
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/test_devel_sections.py:100]: `NotFoundView(browser).is_displayed`
and you don't need navigator here

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        - Creates a new dev portal page in that section
+        - Creates a new dev portal group for that section
+    Test:
+        - Login to dev portal as account with assigned group
+        - Assert that this account has access to this page
+        - Login to dev portal as account without group
+        - Assert that this account hasn't access to this page
+    """
+    custom_devel_login(account=account)
+    browser.url = testconfig["threescale"]["devel"]["url"] + dev_portal_page
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/test_devel_sections.py:93]: `dev_portal_page` should return the whole URL, not just its path. by doing this you don't need to construct whole URL in the test and you can omit `testconfig` fixture

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    view.create(group_name, [dev_portal_section])
+
+    view = navigator.navigate(AccountUserGroupView, account=account)
+    view.update([group_name])
+
+
+@pytest.mark.issue("https://issues.redhat.com/browse/THREESCALE-9020")
+@pytest.mark.issue("https://issues.redhat.com/browse/THREESCALE-836")
+@pytest.mark.skipif("TESTED_VERSION < Version('2.14')")
+def test_dev_portal_sections(account, custom_devel_login, browser, testconfig, dev_portal_section, dev_portal_page,
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/test_devel_sections.py:79]: `dev_portal_group` and `dev_portal_section` are not used in the test

File: testsuite/tests/ui/devel/test_devel_sections.py [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    """Creates a new section in developer portal"""
+    view = navigator.navigate(DeveloperPortalContentNewSectionView)
+    section_name = blame(request, "TestSection")
+    section_path = blame(request, "section")
+    view.create(section_name, section_path, False)
+    section_id = navigator.browser.url.split("/")[-2]
+
+    def cleanup():
+        custom_admin_login()
+        view = navigator.navigate(DeveloperPortalContentEditSectionView, section_id=section_id)
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/test_devel_sections.py:28]: please create an issue where the clean up will be done via API

File: testsuite/tests/ui/devel/test_devel_sections.py (hunk 9 above)
Comment (JaurbanRH) [reply testsuite/tests/ui/devel/test_devel_sections.py:93]: I can't get the whole URL from NewPageView.

**Comment by mganisin** [comment]
@dhlavac @jsmolar ping

**Review by jsmolar — APPROVED** [review]

**Review by dhlavac — APPROVED** [review]

**Pull request merged**